File: 1-revisao/Revisao3.java
import javax.swing.JOptionPane;
import java.util.ArrayList;
import java.util.Random;
import java.lang.Integer;

public class Revisao3 {
    public static void main(String[] args) {
        ArrayList<Integer> vetor = new ArrayList<>();

        int quantidade = Integer.parseInt(JOptionPane.showInputDialog(null,"Quantos números quer gerar? "));
        long tempoInicio, tempoFim;
        
        tempoInicio = System.nanoTime();
        popular(vetor, quantidade);
        exibir(vetor);
        tempoFim = System.nanoTime();
        System.out.println("Tempo de processamento: " + (tempoFim - tempoInicio)/1000000 + " ms");
    }

    public static void popular(ArrayList<Integer> v, int n) {
        Random gerador = new Random();
        for (int i = 0; i < n; i++) {
            v.add(gerador.nextInt(100));
        }
    }

    public static void exibir(ArrayList<Integer> v) {
        for (int i = 0; i < v.size(); i++) {
            System.out.println(v.get(i));
        }
    }
}

File: 1-revisao/revisaoCPP/revisao3.cpp
#include <iostream>
#include <time.h>
#include <stdlib.h>

using namespace std;

int * popular(int *v, int n) {
    //criou um vetor com n posições
    v = (int*)malloc(sizeof(int*) * n);

    //colocou números aleatórios no vetor
    srand(time(NULL));
    for (int i = 0; i < n; i++) {
        v[i] = rand() % 100;
    }
    return v;
}

void exibir(int *v, int n) {
    for (int i = 0; i < n; i++) {
        cout << v[i] << endl;
    }
}

int main() {
    int *vetor;
    int quantidade;
    clock_t tempoInicio, tempoFim;

    cout << "Quantos números deseja gerar? ";
    cin >> quantidade;

    //temporizar
    tempoInicio = clock();
    vetor = popular(vetor, quantidade);
    // exibir(vetor, quantidade);
    tempoFim = clock();
    cout << "Tempo de processamento: "<< (float)(tempoFim - tempoInicio)/CLOCKS_PER_SEC << endl;
}